plugins/optimization-detective/tests/test-class-od-link-collection.php

```diff
@@ -335,6 +335,57 @@ public function data_provider_to_test_add_link(): array {
 				'expected_count'  => 0,
 				'error'           => 'Maximum width must be greater than zero and greater than the minimum width.',
 			),
+			'international_domain_name'                  => array(
+				'links_args'      => array(
+					array(
+						array(
+							'rel'  => 'preload',
+							'href' => 'https://xn--fsq.com/תמונה.jpg',
```

Comment:
Can you add a non-punycode example as well? So instead of `https://xn--fsq.com/תמונה.jpg` this could be:

```suggestion
							'href' => 'https://例.com/תמונה.jpg',
```

But also, let's use IDN subdomains for canonical subdomains. So all the `xn--fsq.com` domains being tested here can be replaced with `xn--fsq.example.com`

Reply:
Updated